src/parser.rs

```diff
@@ -414,7 +416,15 @@ impl Parser<'_> {
                 let pattern_location = self.location;
                 let pattern = self.parse_string()?;
                 let regex = match Regex::new(&pattern) {
-                    Ok(regex) => regex,
+                    Ok(regex) => {
+                        if let Some(_) = regex.captures("") {
```

Comment:
This won't prevent _all_ empty matches, but should suffice for the vast majority, so I think it's probably fine.

(For instance, the regex `"^\b"` matches an empty word boundary at the beginning of the string. This _won't_ match the empty string, but will produce an empty match on, say, the string `"a"`)

Reply:
Interesting example :), I did not think about that. For this the check in `evaluate` is still useful then. I've adapted the execution test to cover this case.

Reply:
Unfortunately Rust regular expressions don't tell you if they are nullable, so we'll have to live with most caught statically, and some dynamically.